code/modules/projectiles/updated_projectiles/gun_helpers.dm

```diff
@@ -583,6 +583,16 @@ should be alright.
 		return
 	src = G
 
+	if(flags_gun_features & GUN_HAS_FULL_AUTO) //No burst on full auto for now.
```

Comment:
You can attach verbs to objects during runtime, the component should handle this instead

Reply:
This has been changed to a toggle between all the fire modes. It is also pending a slight tweak, but the value should stay in the gun if there are possible values unrelated to the automatic fire mode.